server/src/proxyManagement.mjs

```diff
@@ -21,97 +21,114 @@
 import { parse } from 'url';
 import WebSocket from 'ws';
 
-let ws = null
-let responseObject = {}
+const wsMap = new Map();
+const wsMsgMap = new Map();
 
-async function initialize(callback, returnWS) {
-  if(ws) {
-    return true
+async function initializeAndSendRequest(returnWs, command) {
```

Comment:
It seems a bit odd to combine 'initialize' and 'sendRequest'. Like... if you're going to sendRequest multiple times, it doesn't seem right that you also want to initialize each time. If the initialize is really "initialize if necessary" then maybe just don't mention it in the name of the function.

Reply:
You are correct in stating initialize is really an "initialize if necessary". Per your request, I have updated the function to be named `sendRequest()`.